### ISSUE TITLE
JetBrains client downloads every time I connect to the same workspace

### ISSUE THREAD

**Issue by sharkymark** [coder/jetbrains-coder#302]
Why does the JetBrains client download each time I connect Gateway and the Coder plugin to a workspace?

I'm using the new functionality to open Gateway from a Coder workspace button.

[See my sample template](https://github.com/sharkymark/v2-templates/blob/main/pod-idea-icon/main.tf)

![image](https://github.com/coder/jetbrains-coder/assets/2022166/221d29eb-971e-4b7e-b595-2bfca307bfce)

@code-asher

**Comment by code-asher** [comment]
Good question.  This is just a guess, but I wonder if having `ide_download_link` alone forces it to always download?  Maybe we have to specify `ide_path_on_host` as well.  I will try to find some time to experiment.

**Comment by matifali** [comment]
@code-asher our [JetBrains Gateway module](https://registry.coder.com/modules/jetbrains-gateway?tab=source) uses `ide_download_link` and it does not cause redownloading every time.

**Comment by code-asher** [comment]
In that case, in this reproduction is it using the same IDE version?  If it is a different version each time (maybe a newer one), I could see it downloading a new client each time.  Or is this happening even when running the same IDE versions?

**Comment by code-asher** [comment]
Oh wait sorry the issue says this happens when using the Coder Gateway button.  I checked both templates but they seem to be doing more or less the same thing so I am not sure why one would redownload and the other would not.

**Comment by matifali** [comment]
I am not able to reproduce it while using our gateway module.

**Comment by code-asher** [comment]
Closing for now, it does download a new client whenever a new backend is downloaded, but I have not been able to reproduce redownloading when launching the same backend. 

Also, the screenshot actually shows the backend being downloaded, not the client, which might be expected if it is a new workspace or new version.

I have seen one case where JetBrains will download the backend again even if it was already installed, if it was installed in a non-standard location.  I committed a fix for that though (replaced their download mechanism with our own, basically).
